File: src/main/java/com/dongtronic/keeptalking/ManualVersion.java
package com.dongtronic.keeptalking;

public enum ManualVersion {
    CODE241V1("241 version 1", 23, 4),
    MANUAL("Other", 0, 0);

    private String name;
    private int pages;
    private int introPages;

    ManualVersion(String name, int pages, int introPages) {
        this.name = name;
        this.pages = pages;
        this.introPages = introPages;
    }

    public String getName() {
        return name;
    }

    public int getPages() {
        return pages;
    }

    public int getIntroPages() {
        return introPages;
    }

    public static ManualVersion fromName(String name) {
        for (ManualVersion version : ManualVersion.values()) {
            if (version.getName().equalsIgnoreCase(name)) {
                return version;
            }
        }
        return null;
    }
}

File: src/main/java/com/dongtronic/keeptalking/ManualVersion.kt
package com.dongtronic.keeptalking

enum class ManualVersion private constructor(val manualName: String, val pages: Int, val introPages: Int, var filename: String) {
    CODE241V1("241 version 1", 23, 4, "manual-241-v1.pdf"),
    MANUAL("Other", 0, 0, "");


    companion object {

        fun fromName(name: String): ManualVersion? {
            for (version in ManualVersion.values()) {
                if (version.manualName.equals(name, ignoreCase = true)) {
                    return version
                }
            }
            return null
        }
    }
}
